Tests/IteratorsTests/PeekingIteratorTests.swift

```diff
@@ -13,4 +13,13 @@ final class PeekingIteratorTests: XCTestCase {
         XCTAssertEqual(nil, iterator.peek())
         XCTAssertEqual(nil, iterator.next())
     }
+
+    func testPeekingIterator_WithoutPeek() {
+        // Make sure that iterator does not require peek to be called first.
+        var iterator = [1,2,3].makeIterator().peekable()
+        XCTAssertEqual(1, iterator.next())
+        XCTAssertEqual(2, iterator.next())
+        XCTAssertEqual(3, iterator.next())
+        XCTAssertEqual(nil, iterator.next())
+    }
```

Comment:
Let's add one more test that calls `next` first as well as `peek at some point.  Something like

```swift
func testCallingNextFirstAndPeekAtSomePoint() {
    var iterator = [1, 2, 3].makeIterator().peekable()
    XCTAssertEqual(1, iterator.next())
    XCTAssertEqual(2, iterator.peek())
    XCTAssertEqual(2, iterator.next())
    XCTAssertEqual(3, iterator.next())
    XCTAssertEqual(nil, iterator.next())
}
```

Reply:
Done.